src/contracts/middleware/BLSIndexRegistryCoordinator.sol

```diff
@@ -0,0 +1,163 @@
+    }
+
+    /**
+     * @notice Registers the operator with the middleware
+     * @param quorumNumbers are the bytes representing the quorum numbers that the operator is registering for
+     * @param pubkey is the BLS public key of the operator
+     */
+    function registerOperator(bytes calldata quorumNumbers, BN254.G1Point memory pubkey) external {
+        _registerOperator(msg.sender, quorumNumbers, pubkey);
+    }
+
+    function deregisterOperator(address, bytes32, bytes calldata) external override pure {
+        revert("BLSIndexRegistryCoordinator.deregisterOperator: cannot use overrided StakeRegistry.deregisterOperator on BLSIndexRegistryCoordinator");
+    }
+    
+    /**
+     * @notice Deregisters the operator from the middleware
+     * @param deregistrationData is the the data that is decoded to get the operator's deregisteration information
+     */
+    function deregisterOperator(bytes calldata deregistrationData) external {
+        // get the operator's deregisteration information
+        (BN254.G1Point memory pubkey, uint32[] memory quorumToOperatorListIndexes, uint32 globalOperatorListIndex) 
+            = abi.decode(deregistrationData, (BN254.G1Point, uint32[], uint32));
+        // call internal function to deregister the operator
+        _deregisterOperator(msg.sender, pubkey, quorumToOperatorListIndexes, globalOperatorListIndex);
+    }
+
+    /**
+     * @notice Deregisters the operator from the middleware
+     * @param pubkey is the BLS public key of the operator
+     * @param quorumToOperatorListIndexes is the list of the operator's indexes in the quorums they are registered for in the IndexRegistry
+     * @param globalOperatorListIndex is the operator's index in the global operator list in the IndexRegistry
+     */
+    function deregisterOperator(BN254.G1Point memory pubkey, uint32[] memory quorumToOperatorListIndexes, uint32 globalOperatorListIndex) external {
+        _deregisterOperator(msg.sender, pubkey, quorumToOperatorListIndexes, globalOperatorListIndex);
+    }
+
+    function _registerOperator(address operator, bytes calldata quorumNumbers, BN254.G1Point memory pubkey) internal {
+        // TODO: check that the sender is not already registered
+
+        // get the quorum bitmap from the quorum numbers
+        uint256 quorumBitmap = BytesArrayBitmaps.orderedBytesArrayToBitmap_Yul(quorumNumbers);
+        require(quorumBitmap != 0, "BLSIndexRegistryCoordinator: quorumBitmap cannot be 0");
+
+        // register the operator with the BLSPubkeyRegistry and get the operatorId (in this case, the pubkeyHash) back
+        bytes32 operatorId = blsPubkeyRegistry.registerOperator(operator, quorumNumbers, pubkey);
+
+        // register the operator with the IndexRegistry
+        indexRegistry.registerOperator(operatorId, quorumNumbers);
+
+        // register the operator with the StakeRegistry
+        _registerOperator(operator, operatorId, quorumNumbers);
```

Comment:
this `_registerOperator` function and the `_registerOperator` function that it lives inside feel like they need to have different names.
As-is this is way too confusing / feels recursive even though the functions have different inputs.

Maybe `_registerOperatorInAllRegistries` or something for this function?

Reply:
totally, didn't even realize this. here's my proposed fix: https://github.com/Layr-Labs/eigenlayer-contracts/commit/0c907f468e08792b5d0db933ba1a12c64c4c9d32